hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/defaultserializers/JavaDefaultSerializers.java

```diff
@@ -470,21 +476,24 @@ public void write(ObjectDataOutput out, HazelcastJsonValue object) throws IOExce
         public HazelcastJsonValue read(ObjectDataInput in) throws IOException {
             return new HazelcastJsonValue(in.readString());
         }
+    }
+
+    public static final class ByteBufferSerializer implements ByteArraySerializer<ByteBuffer> {
 
         @Override
         public int getTypeId() {
-            return JAVASCRIPT_JSON_SERIALIZATION_TYPE;
+            return JAVA_DEFAULT_BYTE_BUFFER;
         }
-    }
-
-    private abstract static class SingletonSerializer<T> implements StreamSerializer<T> {
```

Comment:
@vbekiaris I am not sure if `SingletonSerializer` was completely useless. cff3d088c63123b21e0891a6eaf7254db868d400 is not fully clear to me: if we should remove `destroy` from all serializers and from the interface because it is empty, or only empty implementations of `destroy` method (but that could be done immediately if the default was also empty).

Reply:
The point in cff3d08 is that empty implementations of `destroy` method can be removed - the reason it wasn't done at that point was that it broke our JUnit compatibility tests and we should wait one more release cycle to do that. That's my understanding of that commit.

Reply:
What was the reason of existince of SingletonSerializer anyway? Deleting it should be safe right?

Reply:
from @vbekiaris comment I understand it should be safe to remove SingletonSerializer